gobblin-modules/gobblin-http/src/main/java/gobblin/http/AvroApacheHttpJoinConverter.java

```diff
@@ -0,0 +1,110 @@
+package gobblin.http;
+
+import java.net.URI;
+import java.util.Map;
+
+import org.apache.avro.Schema;
+import org.apache.avro.generic.GenericData;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.http.HttpHeaders;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpUriRequest;
+import org.apache.http.client.methods.RequestBuilder;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.HttpClientBuilder;
+
+import com.typesafe.config.Config;
+
+import lombok.extern.slf4j.Slf4j;
+
+import gobblin.async.BufferedRecord;
+import gobblin.config.ConfigBuilder;
+import gobblin.configuration.WorkUnitState;
+import gobblin.utils.HttpConstants;
+import gobblin.utils.HttpUtils;
+
+
+/**
+ * Apache version of http join converter
+ */
+@Slf4j
+public class AvroApacheHttpJoinConverter extends AvroHttpJoinConverter<HttpUriRequest, CloseableHttpResponse> {
+  @Override
+  public HttpClient<HttpUriRequest, CloseableHttpResponse> createHttpClient(WorkUnitState workUnit)
+      throws IllegalStateException {
+    Config config = ConfigBuilder.create().loadProps(workUnit.getProperties(), CONF_PREFIX).build();
+    return new ApacheHttpClient(HttpClientBuilder.create(), config, workUnit.getTaskBroker());
+  }
+
+  @Override
+  public ResponseHandler<CloseableHttpResponse> createResponseHandler(WorkUnitState workUnit)
+      throws IllegalStateException {
+    return new HttpResponseHandler();
+  }
+
+  @Override
+  protected ConverterRequestBuilder<HttpUriRequest> createRequestBuilder(WorkUnitState workUnit)
+      throws IllegalStateException {
+    return new ConverterApacheHttpRequestBuilder(workUnit);
+  }
+
+  @Override
+  public void fillHttpOutputData(Schema httpOutputSchema, GenericRecord outputRecord, HttpUriRequest rawRequest,
+      CloseableHttpResponse response, ResponseStatus status) {
+    // TODO: add response to the output record
+    GenericRecord httpOutput = new GenericData.Record(httpOutputSchema);
+
+    httpOutput.put("urlRequest", null);
+
+    outputRecord.put("httpOutput", httpOutput);
+  }
+
+  public static class ConverterApacheHttpRequestBuilder extends ConverterRequestBuilder<HttpUriRequest> {
```

Comment:
@ibuenros HttpRequestBuilder assumes template is type GenericRecord, but HttpOperation here is "NOT EXACTLY" generic type because it depends on the avro version.